hrreport/CRM/HRReport/Upgrader.php

```diff
@@ -255,4 +234,34 @@ public function upgrade_1403() {
     }
     return TRUE;
   }
+
+  /**
+   * Removes unused absence report
+   *
+   * @return bool
+   */
+  public function upgrade_1404() {
+    $templates = [
+      'CRM_HRReport_Form_Activity_HRAbsence',
+      'CRM_HRReport_Form_Activity_HRAbsenceCalendar',
+      'CRM_HRReport_Form_Activity_HRAbsenceDates'
+    ];
+    $reports = civicrm_api3('OptionValue', 'get', [
+      'return' => ['name'],
+      'option_group_id' => 'report_template',
+      'name' => ['IN' => $templates],
+    ]);
+
+    if ($reports['count'] > 0) {
+      $templateIds = array_column(array_values($reports['values']), 'id');
+      foreach ($templateIds as $templateId) {
+        civicrm_api3('OptionValue', 'delete', ['id' => $templateId]);
+      }
+
+      $sql = "DELETE FROM `civicrm_managed` WHERE `entity_id` IN ('" . implode("','", $templateIds) . "')";
```

Comment:
The `entity_id` field is not unique. This query will delete more than you want. Also, keep in mind that the `ReportInstance` instances are also managed entities, so you'll need to remove them here as well